soh/soh/Enhancements/randomizer/hook_handlers.cpp

```diff
@@ -1483,6 +1574,284 @@ void RandomizerOnActorInitHandler(void* actorRef) {
+                Actor_Delete(&gPlayState->actorCtx, actor, gPlayState);
+            }
+            //Special case for Phantom Ganon's horse (and fake), as they're considered "background actors",
+            //but still control the boss fight flow.
+            if (!Flags_GetRandomizerInf(RAND_INF_PHANTOM_GANON_SOUL) && actor->id == ACTOR_EN_FHG) {
+                Actor_Delete(&gPlayState->actorCtx, actor, gPlayState);
+            }
+        }
+    }
+
+    // In MQ Spirit, remove the large silver block in the hole as child so the chest in the silver block hallway
+    // can be guaranteed accessible
+    if (
+        actor->id == ACTOR_OBJ_OSHIHIKI &&
+        LINK_IS_CHILD &&
+        IsGameMasterQuest() &&
+        gPlayState->sceneNum == SCENE_SPIRIT_TEMPLE && actor->room == 6 && // Spirit Temple silver block hallway
+        actor->params == 0x9C7 // Silver block that is marked as in the hole
+    ) {
+        Actor_Kill(actor);
+        return;
+    }
+
+    if (
+        // If child is in the adult shooting gallery or adult in the child shooting gallery, then despawn the shooting gallery man
+        actor->id == ACTOR_EN_SYATEKI_MAN &&
+        RAND_GET_OPTION(RSK_SHUFFLE_INTERIOR_ENTRANCES) &&
+        (
+            (LINK_IS_CHILD && Entrance_SceneAndSpawnAre(SCENE_SHOOTING_GALLERY, 0x00)) || //Kakariko Village -> Adult Shooting Gallery, index 003B in the entrance table
+            (LINK_IS_ADULT && Entrance_SceneAndSpawnAre(SCENE_SHOOTING_GALLERY, 0x01)) //Market -> Child Shooting Gallery,              index 016D in the entrance table
+        )
+    ) {
+        Actor_Kill(actor);
+        return;
+    }
+}
+
+void RandomizerOnGameFrameUpdateHandler() {
+    if (Flags_GetRandomizerInf(RAND_INF_HAS_INFINITE_QUIVER)) {
+        AMMO(ITEM_BOW) = CUR_CAPACITY(UPG_QUIVER);
+    }
+
+    if (Flags_GetRandomizerInf(RAND_INF_HAS_INFINITE_BOMB_BAG)) {
+        AMMO(ITEM_BOMB) = CUR_CAPACITY(UPG_BOMB_BAG);
+    }
+
+    if (Flags_GetRandomizerInf(RAND_INF_HAS_INFINITE_BULLET_BAG)) {
+        AMMO(ITEM_SLINGSHOT) = CUR_CAPACITY(UPG_BULLET_BAG);
+    }
+
+    if (Flags_GetRandomizerInf(RAND_INF_HAS_INFINITE_STICK_UPGRADE)) {
+        AMMO(ITEM_STICK) = CUR_CAPACITY(UPG_STICKS);
+    }
+
+    if (Flags_GetRandomizerInf(RAND_INF_HAS_INFINITE_NUT_UPGRADE)) {
+        AMMO(ITEM_NUT) = CUR_CAPACITY(UPG_NUTS);
+    }
+
+    if (Flags_GetRandomizerInf(RAND_INF_HAS_INFINITE_MAGIC_METER)) {
+        gSaveContext.magic = gSaveContext.magicCapacity;
+    }
+
+    if (Flags_GetRandomizerInf(RAND_INF_HAS_INFINITE_BOMBCHUS)) {
+        AMMO(ITEM_BOMBCHU) = 50;
+    }
+
+    if (Flags_GetRandomizerInf(RAND_INF_HAS_INFINITE_MONEY)) {
+        gSaveContext.rupees = CUR_CAPACITY(UPG_WALLET);
+    }
+
+    if (!Flags_GetRandomizerInf(RAND_INF_HAS_WALLET)) {
+        gSaveContext.rupees = 0;
+    }
+}
+
+extern "C" void func_8099485C(DoorGerudo* gerudoDoor, PlayState* play);
+
+void RandomizerOnActorUpdateHandler(void* refActor) {
```

Comment:
This is by far the most expensive hook we can register, we need to be very careful and precise about how and when this is registered. Each of the different target actors should be specified by the `RegisterGameHookForID` hook, and these hooks should only be registered when their respective options are on

Reply:
eg
```cpp
onFileLoad(() {
  // unregister skeletonKeyDoorUpdate

  if (skeletonKeyEnabled) {
    // Because we're doing RegisterGameHookForID this will only ever be called for door updates
    GameInteractor::Instance->RegisterGameHookForID<GameInteractor::OnActorUpdate>(ACTOR_EN_DOOR, skeletonKeyDoorUpdate);
  }
})